### PR TITLE
add bash script to install packages on non-Linux systems

### PR DESCRIPTION
**Pull request by vszakats**
*BSD systems are currently supported and macOS actually tested.

Ref: https://github.com/mxe/mxe/issues/1708

### CODE REVIEW

**Comment by starius** [comment]
Please add the following comment to the file:

```
# This file is part of MXE. See LICENSE.md for licensing information.
```

File: tools/mxe-get [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+    echo "! Installing mxe package '$1'"
+    mxe_get_pkg "$1"
+    shift
+  done
+
+  echo "! Installed:${done//|/}"
+
+  echo '! Retargeting symlinks...'
+  find . -type l -name '*' | while IFS= read -r f; do
+    # TOFIX: readlink may need to be adapted for non-macOS systems
```
Comment (starius) [inline tools/mxe-get]: s/TOFIX/FIXME/

**Comment by starius** [comment]
I tested it under Linux. It failed with the following error:

```
$ ./mxe-get 
! Downloading and verifying mxe APT package list...
gpg: new configuration file `/home/test/.gnupg/gpg.conf' created
gpg: WARNING: options in `/home/test/.gnupg/gpg.conf' are not yet active during this run
gpg: requesting key DB45C6AB from hkps server keyserver.ubuntu.com
gpgkeys: HTTP fetch error 1: unsupported protocol
gpg: no valid OpenPGP data found
```

Fixed by replacing `hkps://keyserver.ubuntu.com` with `x-hkp://keys.gnupg.net`

**Comment by starius** [comment]
I tried to install Lua using the script:
```
$ ./mxe-get mxe-i686-w64-mingw32.static-lua
```

It installed only packages `mxe-i686-w64-mingw32.static-lua` and `mxe-i686-w64-mingw32.static-mxe-conf`. Compiler and other common packages were not installed.

**Comment by vszakats** [comment]
(I've been busy with some updates just while you reviewed)

- Replace full MIT license with the reference to LICENSE.md
- TOFIX &rarr; FIXME done.
- `linux-gnu` packages are excluded. These have no use on non-Linux (f.e. *BSD) systems.
- `mxe-requirements` and `mxe-source` are also excluded. I didn't need them and their naming would be a special case, so it needs to be implemented if important. Also because `mxe-get` is part of `mxe-source`, it needs to present already to run the script.
- As for the GPG error. I use secure protocol to get the keys and this requires either GPG 2.x or the `gnupg-curl` packages to be installed on Linux. It also works fine on macOS with Homebrew `gpg` package installed. `hkp` and `x-hkp` are both cleartext and as you may have noticed :) I avoid it everywhere.

**Comment by starius** [comment]
> are both cleartext and as you may have noticed :) I avoid it everywhere

I noticed this :)

I encourage encrypting everything in general, but in this particular case it introduces maintenance cost, which I'm not a big fan of. To get privacy from my ISP, I'd rather route all the traffic through Tor instead of fixing things like `hkps` one by one. Nevertheless it is up to you to decide which scheme to use in this case.

I installed `gnupg-curl` but it still fails:

```
gpg: requesting key DB45C6AB from hkps server keyserver.ubuntu.com
gpgkeys: HTTP fetch error 60: server certificate verification failed. CAfile: none CRLfile: none
gpg: no valid OpenPGP data found.
```

**Comment by vszakats** [comment]
Went with replacing `gpg` with a `curl | gpg` combo. Let's hope this works more universally.

**Comment by vszakats** [comment]
Added to rewrite hardcoded paths in some filetypes. Also added an option to install Linux packages. And some minor touchups.

**Comment by starius** [comment]
Looks good to me!

**Comment by starius** [comment]
Thanks!

**Pull request merged**